go/vt/schemadiff/analysis.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schemadiff
+
+import "vitess.io/vitess/go/vt/sqlparser"
+
+// AlterTableRotatesRangePartition answers `true` when the given ALTER TABLE statemnts performas any sort
+// of range partition rotation, that is applicable immediately and without moving data.
+// Such would be:
+// - Dropping any partition(s)
+// - Adding a new partition (empty, at the end of the list)
+func AlterTableRotatesRangePartition(alterTable *sqlparser.AlterTable) (bool, error) {
```

Comment:
As we discussed via Slack, I think that we should at least check that this is indeed a RANGE partition change (vs. e.g. LIST).

Reply:
done